tests/chainerx_tests/unit_tests/routines_tests/test_activation.py

```diff
@@ -75,6 +76,53 @@ def forward_xp(self, inputs, xp):
 ]
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product([
+        chainer.testing.from_pytest_parameterize(
+            'shape', [
+                (2, 2),
+                (3, 3, 3),
+                (5, 5, 5),
+                (4, 1, 2, 4)
+            ]),
+        chainer.testing.from_pytest_parameterize(
+            'in_dtypes,out_dtype', math_utils.in_out_dtypes_math_functions)
+    ])
+))
+class TestLeakyRelu(op_utils.ChainerOpTest):
+
+    dodge_nondifferentiable = True
+    slope = 0.2
+
+    def setup(self, float_dtype):
+        dtype = float_dtype
+
+        if float_dtype == 'float16':
+            self.check_forward_options.update({'atol': 1e-4, 'rtol': 1e-3})
+            self.check_backward_options.update({'atol': 5e-2, 'rtol': 5e-2})
+            self.check_double_backward_options.update(
+                {'atol': 1e-2, 'rtol': 5e-2})
+
+        self.dtype = dtype
+
+    def generate_inputs(self):
+        shape = self.shape
+        dtype = self.dtype
+        x = array_utils.create_dummy_ndarray(numpy, shape, dtype)
```

Comment:
I think this deterministic input generation combined with `dodge_nondifferentiable` causes an infinite loop. How about writing this test similar to the ReLU test below?